lib/msal-core/src/ScopeSet.ts

```diff
@@ -1,69 +1,189 @@
+import { StringUtils } from "./utils/StringUtils";
+import { ClientConfigurationError } from "./error/ClientConfigurationError";
+import { ResponseTypes } from "./utils/Constants";
+
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
 export class ScopeSet {
 
+    private scopes: Set<string>;
+    private originalScopes: Set<string>;
+    private clientId: string;
+    private scopesRequired: boolean;
+
+    constructor(inputScopes: Array<string>, appClientId: string, isLoginCall: boolean) {
+        this.scopesRequired = !isLoginCall;
+        this.clientId = appClientId;
+        // Validate and filter scopes (the validate function will throw if validation fails)
+        this.validateInputScopes(inputScopes);
+        if (!inputScopes) {
+            this.scopes = this.originalScopes = new Set<string>(StringUtils.convertArrayEntriesToLowerCase([this.clientId]));
+        } else {
+            this.scopes = this.originalScopes = new Set<string>(StringUtils.convertArrayEntriesToLowerCase(inputScopes));
+        }
+    }
+
     /**
-   * Check if there are dup scopes in a given request
-   *
-   * @param cachedScopes
-   * @param scopes
-   */
-    // TODO: Rename this, intersecting scopes isn't a great name for duplicate checker
-    static isIntersectingScopes(cachedScopes: Array<string>, scopes: Array<string>): boolean {
-        cachedScopes = this.convertToLowerCase(cachedScopes);
-        for (let i = 0; i < scopes.length; i++) {
-            if (cachedScopes.indexOf(scopes[i].toLowerCase()) > -1) {
-                return true;
+     * Factory method to create ScopeSet from string
+     * 
+     * @param inputScopeString 
+     * @param appClientId 
+     * @param isLoginCall 
+     */
+    static fromString(inputScopeString: string, appClientId: string, isLoginCall: boolean): ScopeSet {
```

Comment:
name ? maybe fromSpaceDelimtedString

Reply:
I think fromString is fine here, I am changing the comment to reflect the details of the function. We can change the name later if we find there are issues.